### PR TITLE
Add route specific onRequest hook

### PR DESCRIPTION
**Pull request by jchip**
Redoing https://github.com/OptimalBits/redbird/pull/203 without prettifying.

Sorry about the formatting in README.md my editor did automatically, but looks like it's just stripping trailing blanks at end of lines.  The main changes are adding lines 515-518.

### CODE REVIEW

**Comment by jchip** [comment]
drop node.js 6 and 7 because they fail for CI because I am using Hapi 17 as server for testing and it's using async/await.

File: lib/proxy.js [hunk 1]
```diff
@@ -610,7 +622,7 @@ ReverseProxy.prototype._getTarget = function (src, req) {
       // remove prefix from src
       //
       req._url = url; // save original url
-      req.url = url.substr(pathname.length) || '/';
+      req.url = url.substr(pathname.length) || '';
```
Comment (manast) [inline lib/proxy.js]: why removing the '/' ?

Comment (jchip) [reply lib/proxy.js]: the extra `/` cause 404 error.  `test/test_onrequest.js` fails.

**Comment by jchip** [comment]
thanks for merging

**Pull request merged**